lib/matplotlib/transforms.py

```diff
@@ -682,6 +682,70 @@ def intersection(bbox1, bbox2):
 class Bbox(BboxBase):
     """
     A mutable bounding box.
+
+    Examples
+    --------
+    **Create from known bounds**
+
+    The default constructor takes the boundary "points" ``[[xmin, ymin],
+    [xmax, ymax]]``.
+
+        >>> Bbox([[1, 1], [3, 7]])
+        Bbox([[1.0, 1.0], [3.0, 7.0]])
+
+    Alternatively, a Bbox can be created from the flattened array, the
+    so-called "extents" ``(xmin, ymin, xmax, ymax)``
+
+        >>> Bbox.from_extents(1, 1, 3, 7)
+        Bbox([[1.0, 1.0], [3.0, 7.0]])
+
+    or from the "bounds" ``(xmin, ymin, width, height)``.
+
+        >>> Bbox.from_bounds(1, 1, 2, 6)
+        Bbox([[1.0, 1.0], [3.0, 7.0]])
+
+    **Create from collections of points**
+
+    The "empty" object for accumulating Bboxs is the null bbox, which has
+    ``(min, max) = (np.inf, -np.inf)`` for each dimension.
+
+        >>> Bbox.null()
+        Bbox([[inf, inf], [-inf, -inf]])
+
+    Adding points to the null bbox will give you the bbox of those points.
+
+        >>> box = Bbox.null()
+        >>> box.update_from_data_xy([[1, 1]])
+        >>> box
+        Bbox([[1.0, 1.0], [1.0, 1.0]])
+        >>> box.update_from_data_xy([[2, 3], [3, 2]], ignore=False)
+        >>> box
+        Bbox([[1.0, 1.0], [3.0, 3.0]])
+
+    Setting ``ignore=True`` is equivalent to starting over from a null bbox.
+
+        >>> box.update_from_data_xy([[1, 1]], ignore=True)
+        >>> box
+        Bbox([[1.0, 1.0], [1.0, 1.0]])
+
+    .. warning:: Not setting ``ignore`` can have unexpected results.
+
+    **Surprising properties of ``null``**
```

Comment:
Please don't use a judgemental heading. Surprising is subjective, and also we as a library should not claim to have a surprising API without commenting on that.

Something like this is more factual and clear:

~~~
**Properties of the ``null`` bbox**

.. note::
    The current behavior of `Bbox.null()` may be surprising as it does not
    behave like a "zero" object in a mathematical sense. We may change
    that in the future (with a deprecation period).
~~~